test/embedding/embedtest.cc

```diff
@@ -153,19 +154,26 @@ int RunNodeInstance(MultiIsolatePlatform* platform,
     Context::Scope context_scope(setup->context());
 
     MaybeLocal<Value> loadenv_ret;
-    if (snapshot) {
+    if (snapshot) {  // Deserializing snapshot
       loadenv_ret = node::LoadEnvironment(env, node::StartExecutionCallback{});
-    } else {
+    } else if (is_building_snapshot) {
```

Comment:
Should we sync the code in the embedding docs (https://github.com/nodejs/node/blob/6a489df73b0ab64f8fab016a7a71e209066fdb82/doc/api/embedding.md?plain=1#L112-L166) too?

Reply:
Maybe, the code in there would continue to work, the snapshot capability was added by https://github.com/nodejs/node/pull/45888, so probably out of scope of this PR.

Reply:
The doc has a reference to this file in https://github.com/nodejs/node/blob/6a489df73b0ab64f8fab016a7a71e209066fdb82/doc/api/embedding.md?plain=1#L24, so I thought these were supposed to be in sync but this too lgtm if that's not needed